hal/cpu.h

```diff
@@ -0,0 +1,130 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system kernel
+ *
+ * CPU related routines
+ *
+ * Copyright 2014, 2017, 2018 Phoenix Systems
+ * Author: Jacek Popko, Aleksander Kaminski, Pawel Pisarczyk
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#ifndef _HAL_CPU_H_
+#define _HAL_CPU_H_
+
+#include <arch/cpu.h>
+#include "spinlock.h"
+
+
+/* interrupts */
+
+
+extern void hal_cpuDisableInterrupts(void);
+
+
+extern void hal_cpuEnableInterrupts(void);
+
+
+/* performance */
+
+
+extern void hal_cpuLowPower(time_t us);
+
+/* TODO: it is only related to stm32 - IFDEF in syscall!! To remove! */
```

Comment:
No - for now it is, but we might want to add low power modes on others archs

Reply:
In such case, I can assume that this function will be available in all platforms and I can remove `ifdef` in syscall:
https://github.com/phoenix-rtos/phoenix-rtos-kernel/blob/5e3e16d27303d392620b1203e26166d7718b596d/syscalls.c#L653-L658